container-core/src/test/java/com/yahoo/container/handler/metrics/PrometheusHandlerTest.java

```diff
@@ -0,0 +1,131 @@
+    private static final String CUSTOM_CONSUMER = "custom-consumer";
+
+    private static RequestHandlerTestDriver testDriver;
+
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule(options().dynamicPort());
+
+    @Before
+    public void setup() {
+        setupWireMock();
+        var handler = new PrometheusHandler(Executors.newSingleThreadExecutor(),
+                new MetricsProxyApiConfig.Builder()
+                        .prometheusPort(wireMockRule.port())
+                        .prometheusApiPath(MOCK_METRICS_PATH)
+                        .metricsPort(0)
+                        .metricsApiPath("/Not/In/Use")
+                        .build());
+        testDriver = new RequestHandlerTestDriver(handler);
+
+    }
+
+    private void setupWireMock() {
+        wireMockRule.stubFor(get(urlPathEqualTo(MOCK_METRICS_PATH))
+                .willReturn(aResponse().withBody(RESPONSE)));
+
+        // Add a slightly different response for a custom consumer.
+        String myConsumerResponse = RESPONSE.replaceAll(CPU_METRIC, REPLACED_CPU_METRIC);
+        wireMockRule.stubFor(get(urlPathEqualTo(MOCK_METRICS_PATH))
+                .withQueryParam("consumer", equalTo(CUSTOM_CONSUMER))
+                .willReturn(aResponse().withBody(myConsumerResponse)));
+    }
+
+    @Test
+    public void v2_response_contains_values_uri() throws Exception {
+        String response = testDriver.sendRequest(V2_URI).readAll();
+        JSONObject root = new JSONObject(response);
+        assertTrue(root.has("resources"));
+
+        JSONArray resources = root.getJSONArray("resources");
+        assertEquals(1, resources.length());
+
+        JSONObject valuesUri = resources.getJSONObject(0);
+        assertEquals(VALUES_URI, valuesUri.getString("url"));
+    }
+
+    @Test
+    public void visually_inspect_values_response() throws Exception {
+        String response = testDriver.sendRequest(VALUES_URI).readAll();
+        System.out.println(response);
+    }
+
+    @Test
+    public void invalid_path_yields_error_response() throws Exception {
+        String response = testDriver.sendRequest(V2_URI + "/invalid").readAll();
+        JSONObject root = new JSONObject(response);
+        assertTrue(root.has("error"));
+        assertTrue(root.getString("error" ).startsWith("No content"));
+    }
+
+    //Currently broken as no way to convert to Prometheus format around here
+    @Ignore
+    @Test
+    public void values_response_is_equal_to_test_file() {
+        String response = testDriver.sendRequest(VALUES_URI).readAll();
+        assertEquals(RESPONSE, response);
+    }
+
+    @Test
+    public void consumer_is_propagated_to_metrics_proxy_api() throws JSONException {
+        String response = testDriver.sendRequest(VALUES_URI + consumerQuery(CUSTOM_CONSUMER)).readAll();
+
+        assertTrue(response.contains(REPLACED_CPU_METRIC));
+    }
+
+    private static String getFileContents(String filename) {
```

Comment:
You can probably remove this method and use the one in MetricsV2HandlerTest, after making it package private. Less code duplication.